src/analysis/AppLauncher.ts

```diff
@@ -218,11 +251,52 @@ const ApplicationLauncher = _.flow(
     });
 
     useEffect(() => {
-      const runtime = getCurrentRuntime(runtimes);
       const runtimeStatus = getConvertedRuntimeStatus(runtime);
 
+      const setupWelder = _.flow(
+        Utils.withBusyState(setBusy),
+        withErrorReporting('Error setting up analysis file syncing')
+      )(async (runtime: Runtime) => {
+        // The special case here is because for GCP, Jupyter and JupyterLab can both be run on the same runtime and a
+        // user may toggle back and forth between them. In order to keep notebooks tidy and in a predictable location on
+        // disk, we mirror the localBaseDirectory used by edit mode for Jupyter.
+        // Once Jupyter is phased out in favor of JupyterLab for GCP, the localBaseDirectory can be '' for all cases
+        const localBaseDirectory =
+          !!googleProject && application === runtimeToolLabels.JupyterLab ? `${workspaceName}/edit` : '';
+
+        const { storageContainerName } = azureContext
+          ? await Ajax(signal).AzureStorage.details(workspaceId)
+          : { storageContainerName: bucketName };
+        const cloudStorageDirectory = azureContext
+          ? `${storageContainerName}/analyses`
+          : `gs://${storageContainerName}/notebooks`;
+
+        googleProject
+          ? await Ajax()
+              .Runtimes.fileSyncing(googleProject, runtime.runtimeName)
+              .setStorageLinks(
+                localBaseDirectory,
+                '',
+                cloudStorageDirectory,
+                getPatternFromRuntimeTool(getToolLabelFromRuntime(runtime))
+              )
+          : await Ajax()
+              .Runtimes.azureProxy(runtime.proxyUrl)
+              .setStorageLinks(
+                localBaseDirectory,
+                cloudStorageDirectory,
+                getPatternFromRuntimeTool(getToolLabelFromRuntime(runtime))
+              );
+      });
+
+      if (shouldSetupWelder && runtime && runtimeStatus === 'Running') {
+        setupWelder(runtime);
+        setShouldSetupWelder(false);
+      }
+
       const computeIframeSrc = withErrorReporting('Error loading application iframe')(async () => {
-        const getSparkInterfaceSource = (proxyUrl) => {
+        const getSparkInterfaceSource = (proxyUrl): string => {
+          const sparkInterface = Nav.getCurrentUrl().href.split('/').pop();
```

Comment:
@LizBaldo this is the line that fixes it. Before, we were getting it through nav parameters here https://github.com/DataBiosphere/terra-ui/pull/4758/files#diff-a48615736ec0d12e4c2bfefed1a3104029f10622d57802f4fece43d2f0a85368L50, but I have no idea what changes to the overall nav logic broke this (since application is still defined). It may also be changes to the `wrapWorkspace` function that broke it. It was a bit of a rabbit hole, with neither having good documentation or examples.

Reply:
Thanks! Sorry I missed that